static/js/simulator.js

```diff
@@ -596,19 +600,19 @@ class General_Word extends Word {
     }
 
     /**
-     * Stores a fixed point number into the word.
+     * Stores a fixed point number into the word.  Be sure to distinguish positive and negative zero.
      *
      * @param {number} number       Number to be stored.
      */
     set fixed_point(number) {
         var sign_bit = "";
```

Comment:
prefer:
`let sign_bit;` - generally better practice to declare the variable without initializing than init to empty string, and let's avoid `var` for its odd scoping behaviors — if you could go through and generally clean up use of var and empty string initialization throughout, that would be great!

Reply:
Question: Can I use empty strings in a case like in the function `binary_add` (line 161), where I concatenate to the string in a loop?

Reply:
Yes—I don't want to be draconian about this principle. In the case of `binary_add`, declaring but not initializing the variable would require you to add some logic in the for loop (checking if it's `undefined` to catch the case of the first run), which would make the loop harder to read and maintain. So, in this case, I think initializing to the empty string is totally fine and, in fact, better.

The more general case is that defining a variable without initializing it lets us make a general assumption about our codebase that we use `=== undefined` to see if we have a value, whereas getting in the habit of `let foo = ""` means we end up with other falsey states to deal with. While it doesn't actually cause problems in any of the instances that you wrote it in, I'd like to develop the habit in this codebase of this practice, which will simplify things in the long term.

Javascript is so permissive that we have to enforce disciplines internally about this kind of stuff to keep from going mad.